### PR TITLE
fix: missing messages after migration

### PR DESCRIPTION
**Pull request by MohamadJaara**
…lf ID

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

missing message after migraiton from scala

### Causes (Optional)

one of the scala versions is setting the domain value as an empty string in the DB instead of null.
the migration code set the domain as the user self domain if the DB version is too old (before federation) but in this case the scala is randomly setting the domain (for some it is correct for others it is an empty string)

### Solutions

for new user:
- check the domain when mapping and user the user self domain in case of null or blank

for efected users:
- the fix will be from kalium in form of DB migration that lookup the corrupted IDs (user and conversations IDs) and fixes it by appending the user self domain to it

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/1702/

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ def defineFlavor() {
     } else if(branchName == "internal") {
         return 'Internal'
     }
-    return 'Staging'
+    return 'Prod'
```
Comment (vitorhugods) [inline Jenkinsfile]:
```suggestion
    return 'Staging'
```

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/1/) **failed**.

**Review by vitorhugods — COMMENTED** [review]
Other than the reversing of the `Staging` flavour, looks good to me.

Any plans on identifying that this failure happened and then retriggering the migration, so users can recover from it?

I'm thinking of persisting a new migration flag, like `wasMigrationRecoveryAttempted` (defaults to false).

When starting the app/checking if migration is needed:
1. Check the `wasMigrationRecoveryAttempted` flag.
  * If it's `true`. Stop this procedure 
2. Querying the Scala DB for empty domains.
  * If there are none. Stop this procedure and mark `wasMigrationRecoveryAttempted` as `true`.
3. Re-perform migration and mark `wasMigrationRecoveryAttempted` as `true`.

**Comment by tmspzz** [comment]
Sorry for being Captain Obvious here: please add test 😉

**Comment by tmspzz** [comment]
> Any plans on identifying that this failure happened and then retriggering the migration, so users can recover from it?

I think @MohamadJaara wanted to write a DB migration that would address the issue for everyone without additional changes to the migration code

**Comment by MohamadJaara** [comment]
@tmspzz we do need both the migration to fix it for effected users and this for new users

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6431765)(6431765)-prod-compatrelease.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/prod/compatrelease/com.wire.android-vv4.1.2(6431765)(6431765)-prod-compatrelease.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6433066)(6433066)-prod-compatrelease.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/prod/compatrelease/com.wire.android-vv4.1.2(6433066)(6433066)-prod-compatrelease.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6433311)(6433311)-prod-compatrelease.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/prod/compatrelease/com.wire.android-vv4.1.2(6433311)(6433311)-prod-compatrelease.apk)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6456567)(6456567)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6456567)(6456567)-staging-compat.apk)

File: app/src/test/kotlin/com/wire/android/migration/MigrationMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        val selfUser = UserId("self_id", "self_domain")
+
+        val expected = QualifiedID(scalaId, selfUser.domain)
+        migrationMapper.toQualifiedId(scalaId, domain, selfUser).also {
+            assertEquals(expected, it)
+        }
+    }
+
+    @Test
+    fun givenNonEmptyStringDomain_whenMappingIdFromScala_thenUserDomain() {
```
Comment (vitorhugods) [inline app/src/test/kotlin/com/wire/android/migration/MigrationMapperTest.kt]:
```suggestion
    fun givenNonEmptyStringDomain_whenMappingIdFromScala_thenKeepTheDomainFromScala() {
```

Comment (vitorhugods) [inline app/src/test/kotlin/com/wire/android/migration/MigrationMapperTest.kt]:
```suggestion
fun givenNonEmptyStringDomain_whenMappingIdFromScala_thenKeepTheDomainFromScala() {
```

File: app/src/main/kotlin/com/wire/android/migration/feature/MigrateActiveAccountsUseCase.kt [hunk 3]
```diff
@@ -104,7 +109,7 @@ class MigrateActiveAccountsUseCase @Inject constructor(
     }
 
     private fun isDataComplete(serverConfig: ServerConfig, activeAccount: ScalaActiveAccountsEntity): Boolean {
-        val isDomainExist = (activeAccount.domain != null) or (serverConfig.metaData.domain != null)
+        val isDomainExist = (!activeAccount.domain.isNullOrBlank()) or (serverConfig.metaData.domain != null)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/migration/feature/MigrateActiveAccountsUseCase.kt]:
```suggestion
        val isDomainPresent = (!activeAccount.domain.isNullOrBlank()) or (serverConfig.metaData.domain != null)
```

**Review by mchenani — APPROVED** [review]

**Review by vitorhugods — APPROVED** [review]
Just a few nitpicks.

Looks great.
Good job finding and fixing this!

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1756/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.2(6458782)(6458782)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.2(6458782)(6458782)-staging-compat.apk)

**Pull request merged**